mono/tests/tailcall/Makefile

```diff
@@ -0,0 +1,54 @@
+# Tests are written in C# to be amenable to tail calls.
+# ildasm + perl + ilasm insert the tail instructions where appropriate.
+# Original failing and edited expected-to-succeed executables are produced.
+
+# NOTE: This one Makefile works with Microsoft nmake and GNU make.
+# They use different conditional syntax, but each can be nested and inverted within the other.
+
+all: check
+
+ifdef _NMAKE_VER:
+!ifdef _NMAKE_VER
+
+#
+# Microsoft nmake on Windows with desktop CLR
+#
+
+RM_F = del 2>nul /f
+ILDASM = ildasm /nobar /out:$@
+ILASM = ilasm /quiet
+
+!else
+else
+
+#
+# GNU/Posix make on Unix with mono
+#
+
+RM_F = rm -f
+ILDASM = ikdasm >$@
+ILASM = ilasm
+MONO ?= mono
+
+endif
+!endif :
+
+check: tailcall1.exe tailcall1-tail.exe
+	-$(MONO) tailcall1.exe
+	$(MONO) tailcall1-tail.exe
+
+clean:
+	$(RM_F) tailcall1-tail.exe tailcall1-tail.il tailcall1.il tailcall1.exe
+
+tailcall1-tail.exe: tailcall1-tail.il
+	$(ILASM) $? /out:$@
+
+tailcall1-tail.il: tailcall1.il tailcall1.pl
+	perl tailcall1.pl < tailcall1.il > $@
```

Comment:
really? the tests should be as simple as possible and you are patching unstable disassembled compiler output with perl script?